gatovid/models.py

```diff
@@ -271,3 +271,55 @@ def as_dict(self) -> Dict[str, str]:
         d = {c.name: getattr(self, c.name) for c in self.__table__.columns}
         del d["user_id"]
         return d
+
+    @staticmethod
+    def get_item_list(item_type: PurchasableType) -> List[Dict[str, str]]:
+        """
+        Dado el tipo de objeto, devuelve la lista de objetos (JSON) con la
+        información como el nombre o el precio.
+        """
+        if item_type == PurchasableType.PROFILE_PIC:
+            return PROFILE_PICS
+        if item_type == PurchasableType.BOARD:
+            return BOARDS
+
+    def get_price(self) -> int:
+        """
+        Devuelve el precio actual de esta compra. NOTE: no se tienen en cuenta
+        los precios anteriores si el precio del objeto cambia.
+        """
+        item_list = self.get_item_list(self.type)
+        return item_list[self.item_id]["cost"]
+
+    @validates("item_id", "type")
+    def validate_item_id(self, key: str, val) -> None:
+        # Necesitamos validar los dos porque item_id depende del tipo.  Aún así,
+        # no podemos validar los dos de vez, se hace "iterativamente".
+
+        # Evitamos valores vacíos de entrada
+        if val is None:
+            raise InvalidModelException(f"Parámetro {key} vacío")
+
+        if key == "item_id":
+            try:
+                val = int(val)
+            except ValueError:
+                raise InvalidModelException("El ID de objeto debe ser un número")
+        else:
+            try:
+                val = PurchasableType(val)
+            except ValueError:
+                raise InvalidModelException("Tipo de objeto inválido")
+
+        # Hacemos la comprobación en las dos iteraciones
+        item_id = val if key == "item_id" else self.item_id
+        item_type = val if key == "type" else self.type
+        if item_id is None or item_type is None:
```

Comment:
si quieres ser ultra pitonico puedes usar `if None in (item_id, item_type)`

Reply:
cierto siempre se me olvida jajaja